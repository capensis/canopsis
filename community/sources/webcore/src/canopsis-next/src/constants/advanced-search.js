export const ADVANCED_SEARCH_ITEM_TYPES = {
  field: 'field',
  condition: 'condition',
  value: 'value',
  union: 'union',
};

export const ADVANCED_SEARCH_NEXT_ITEM_TYPES = {
  [ADVANCED_SEARCH_ITEM_TYPES.field]: ADVANCED_SEARCH_ITEM_TYPES.condition,
  [ADVANCED_SEARCH_ITEM_TYPES.condition]: ADVANCED_SEARCH_ITEM_TYPES.value,
  [ADVANCED_SEARCH_ITEM_TYPES.value]: ADVANCED_SEARCH_ITEM_TYPES.union,
  [ADVANCED_SEARCH_ITEM_TYPES.union]: ADVANCED_SEARCH_ITEM_TYPES.field,
};

export const ADVANCED_SEARCH_NOT = 'NOT';

export const ADVANCED_SEARCH_UNION_CONDITIONS = {
  and: 'AND',
  or: 'OR',
};

export const ADVANCED_SEARCH_CONDITIONS = {
  less: '<',
  more: '>',
  equal: '=',
  notEqual: '!=',
  like: 'LIKE',
  contains: 'CONTAINS',
};

export const ADVANCED_SEARCH_UNION_REGEXP_PATTERN = new RegExp(`\\s(${Object.values(ADVANCED_SEARCH_UNION_CONDITIONS).join('|')})(\\s|$)`, 'gi');

export const ADVANCED_SEARCH_UNION_FIELDS = [
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.and,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.and,
  },
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.or,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.or,
  },
];
