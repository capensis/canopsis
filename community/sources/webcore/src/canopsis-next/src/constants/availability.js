import { QUICK_RANGES } from './common';

export const AVAILABILITY_SHOW_TYPE = {
  percent: 0,
  duration: 1,
};

export const AVAILABILITY_QUICK_RANGES = {
  [QUICK_RANGES.last1Hour.value]: QUICK_RANGES.last1Hour,
  [QUICK_RANGES.last24Hour.value]: QUICK_RANGES.last24Hour,
  [QUICK_RANGES.today.value]: QUICK_RANGES.today,
  [QUICK_RANGES.yesterday.value]: QUICK_RANGES.yesterday,
  [QUICK_RANGES.last7Days.value]: QUICK_RANGES.last7Days,
  [QUICK_RANGES.previousWeek.value]: QUICK_RANGES.previousWeek,
  [QUICK_RANGES.previousMonth.value]: QUICK_RANGES.previousMonth,
  [QUICK_RANGES.thisMonth.value]: QUICK_RANGES.thisMonth,
  [QUICK_RANGES.thisMonthSoFar.value]: QUICK_RANGES.thisMonthSoFar,
  [QUICK_RANGES.last3Months.value]: QUICK_RANGES.last3Months,
  [QUICK_RANGES.last6Months.value]: QUICK_RANGES.last6Months,
};
