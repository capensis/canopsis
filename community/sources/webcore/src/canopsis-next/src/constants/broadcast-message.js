export const DEFAULT_BROADCAST_MESSAGE_COLOR = '#e75e40';

export const BROADCAST_MESSAGES_STATUSES = {
  active: 0,
  pending: 1,
  expired: 2,
};
