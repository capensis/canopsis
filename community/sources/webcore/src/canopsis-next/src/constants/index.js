export * from './recurrence-rule';
export * from './common';
export * from './entity';
export * from './alarm';
export * from './broadcast-message';
export * from './date';
export * from './dynamic-info';
export * from './event-filter';
export * from './pattern';
export * from './healthcheck';
export * from './idle-rule';
export * from './junit';
export * from './modal';
export * from './pbehavior';
export * from './permission';
export * from './remediation';
export * from './scenario';
export * from './snmp';
export * from './stats';
export * from './store';
export * from './widget';
export * from './kpi';
export * from './user';
export * from './service-weather';
export * from './map';
export * from './flowchart';
export * from './parameters';
export * from './declare-ticket';
export * from './link-rule';
export * from './external-data';
export * from './tag';
export * from './counter';
export * from './calendar';
export * from './icon';
export * from './availability';
