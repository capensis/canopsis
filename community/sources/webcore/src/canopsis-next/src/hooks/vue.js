import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
