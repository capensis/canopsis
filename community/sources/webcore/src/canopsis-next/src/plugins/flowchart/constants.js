export const DIRECTIONS = {
  northWest: 'nw',
  north: 'n',
  northEast: 'ne',
  east: 'e',
  southEast: 'se',
  south: 's',
  southWest: 'sw',
  west: 'w',
};
