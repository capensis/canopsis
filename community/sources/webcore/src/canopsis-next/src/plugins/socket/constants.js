export const REQUEST_MESSAGES_TYPES = {
  ping: 0,
  join: 1,
  leave: 2,
  authenticate: 3,
};

export const RESPONSE_MESSAGES_TYPES = {
  pong: 0,
  ok: 1,
  error: 2,
  close: 3,
  authenticated: 4,
};

export const ICONS_RESPONSE_MESSAGES_TYPES = {
  create: 0,
  update: 1,
  delete: 2,
};

export const MAX_RECONNECTS_COUNT = 10;

export const PING_INTERVAL = 5000;

export const RECONNECT_INTERVAL = 5000;

export const EVENTS_TYPES = {
  customClose: 'custom-close',
  closeRoom: 'close-room',
  networkError: 'network-error',
};

export const ERROR_MESSAGES = {
  authenticationFailed: 'authentication failed',
};
