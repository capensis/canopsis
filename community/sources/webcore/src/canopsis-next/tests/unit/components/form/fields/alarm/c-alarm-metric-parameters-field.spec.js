import { generateShallowRenderer, generateRenderer } from '@unit/utils/vue';

import { createSelectInputStub } from '@unit/stubs/input';
import { ALARM_METRIC_PARAMETERS } from '@/constants';

import CAlarmMetricParametersField from '@/components/forms/fields/kpi/c-alarm-metric-parameters-field.vue';

const stubs = {
  'v-select': createSelectInputStub('v-select'),
};

describe('c-alarm-metric-parameters-field', () => {
  const factory = generateShallowRenderer(CAlarmMetricParametersField, { stubs });
  const snapshotFactory = generateRenderer(CAlarmMetricParametersField);

  it('Value changed after trigger the input', () => {
    const wrapper = factory({
      propsData: {
        value: [],
      },
    });
    const newValue = [ALARM_METRIC_PARAMETERS.ratioInstructions];
    const selectElement = wrapper.find('select.v-select');

    selectElement.vm.$emit('input', newValue);

    const inputEvents = wrapper.emitted('input');

    expect(inputEvents).toHaveLength(1);

    const [eventData] = inputEvents[0];
    expect(eventData).toBe(newValue);
  });

  it('Renders `c-alarm-metric-parameters-field` with default props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        value: [ALARM_METRIC_PARAMETERS.createdAlarms],
      },
    });

    const menuContent = wrapper.findMenu();

    expect(wrapper.element).toMatchSnapshot();
    expect(menuContent.element).toMatchSnapshot();
  });

  it('Renders `c-alarm-metric-parameters-field` with custom props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        value: [ALARM_METRIC_PARAMETERS.createdAlarms, ALARM_METRIC_PARAMETERS.ratioInstructions],
        min: 2,
        name: 'customName',
        hideDetails: true,
        parameters: [
          ALARM_METRIC_PARAMETERS.createdAlarms,
          ALARM_METRIC_PARAMETERS.ratioInstructions,
          ALARM_METRIC_PARAMETERS.ratioTickets,
        ],
        disabledParameters: [
          ALARM_METRIC_PARAMETERS.ratioTickets,
        ],
      },
    });

    const menuContent = wrapper.findMenu();

    expect(wrapper.element).toMatchSnapshot();
    expect(menuContent.element).toMatchSnapshot();
  });

  it('Renders `c-alarm-metric-parameters-field` with all values', () => {
    const wrapper = snapshotFactory({
      propsData: {
        value: Object.values(ALARM_METRIC_PARAMETERS),
      },
    });

    const menuContent = wrapper.findMenu();

    expect(wrapper.element).toMatchSnapshot();
    expect(menuContent.element).toMatchSnapshot();
  });
});
