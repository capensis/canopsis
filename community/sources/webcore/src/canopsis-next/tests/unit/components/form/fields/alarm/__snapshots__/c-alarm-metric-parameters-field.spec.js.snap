// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-alarm-metric-parameters-field Renders \`c-alarm-metric-parameters-field\` with custom props 1`] = `
<div class="v-menu__content theme--light v-autocomplete__content" style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
  <div class="v-select-list v-card theme--light">
    <div role="list" class="v-list theme--light">
      <div disabled="disabled" role="listitem" class="v-list--disabled"><a disabled="disabled" class="v-list__tile v-list__tile--link v-list__tile--disabled v-list__tile--active theme--light primary--text">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="true" disabled="disabled" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple primary--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light primary--text">check_box</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of created alarms</div>
          </div>
        </a></div>
      <div disabled="disabled" role="listitem" class="v-list--disabled"><a disabled="disabled" class="v-list__tile v-list__tile--link v-list__tile--disabled v-list__tile--active theme--light primary--text">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="true" disabled="disabled" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple primary--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light primary--text">check_box</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of alarms with auto remediation</div>
          </div>
        </a></div>
      <div disabled="disabled" role="listitem" class="v-list--disabled"><a disabled="disabled" class="v-list__tile v-list__tile--link v-list__tile--disabled theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox v-input--is-disabled theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" disabled="disabled" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of alarms with tickets created</div>
          </div>
        </a></div>
    </div>
  </div>
</div>
`;

exports[`c-alarm-metric-parameters-field Renders \`c-alarm-metric-parameters-field\` with default props 1`] = `
<div class="v-menu__content theme--light v-autocomplete__content" style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
  <div class="v-select-list v-card theme--light">
    <div role="list" class="v-list theme--light">
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of active alarms</div>
          </div>
        </a></div>
      <div disabled="disabled" role="listitem" class="v-list--disabled"><a disabled="disabled" class="v-list__tile v-list__tile--link v-list__tile--disabled v-list__tile--active theme--light primary--text">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="true" disabled="disabled" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple primary--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light primary--text">check_box</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of created alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of non-displayed alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of alarms under automatic remediation</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of alarms with PBehavior</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of alarms with correlation</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Total number of acks</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of canceled acks</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of active alarms with acks</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of active alarms with tickets</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Number of active alarms without tickets</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of correlated alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of alarms with auto remediation</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of alarms with tickets created</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of non-displayed alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">% of manually remediated alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Average time to ack alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Average time to resolve alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Minimum time to ack alarms</div>
          </div>
        </a></div>
      <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
          <div class="v-list__tile__action">
            <div class="v-input v-input--selection-controls v-input--checkbox theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="checkbox" type="checkbox" value="">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">Maximum time to ack alarms</div>
          </div>
        </a></div>
    </div>
  </div>
</div>
`;

exports[`c-alarm-metric-parameters-field Renders \`c-alarm-metric-parameters-field\` with external metrics 1`] = `
<div class="v-menu__content theme--light v-autocomplete__content" style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
  <div class="v-select-list v-card theme--light">
    <div role="list" class="v-list theme--light">
      <div role="listitem">
        <div class="v-list__tile theme--light">
          <div class="v-list__tile__content">
            <div class="v-list__tile__title">No data available</div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
